objectivec/GPBWellKnownTypes.m

```diff
@@ -115,3 +115,46 @@ - (void)setTimeIntervalSince1970:(NSTimeInterval)timeIntervalSince1970 {
 }
 
 @end
+
+NSString *const GPBTypeGoogleApisComPrefix = @"type.googleapis.com/";
+
+@implementation GPBAny (GBPWellKnownTypes)
+
+- (NSString*)getAnyTypeName {
+  NSAssert([self.typeURL hasPrefix:GPBTypeGoogleApisComPrefix],
+           @"Invalid any type url (%@).", self.typeURL);
+  return [self.typeURL substringFromIndex:[GPBTypeGoogleApisComPrefix length]];
+}
+
+- (instancetype)initWithMessage:(GPBMessage*)message {
+  self = [super init];
+  if (self) {
+    [self setMessage:message];
+  }
+  return self;
+}
+
+- (void)setMessage:(GPBMessage*)message {
+  self.typeURL = [GPBTypeGoogleApisComPrefix stringByAppendingString:message.descriptor.name];
+  self.value = message.data;
+}
+
+- (GPBMessage*)messageOfClass:(Class)messageClass {
+  if ([[self getAnyTypeName] isEqualToString:messageClass.descriptor.name]) {
+    GPBMessage* message = [messageClass message];
+    [message mergeFromData:self.value extensionRegistry:nil];
+    return message;
+  } else {
+    return nil;
+  }
+}
+
+- (BOOL)wrapsMessageOfClass:(Class)messageClass {
+  NSAssert([messageClass isSubclassofClass:[GPBMessage class]],
```

Comment:
Keeping the software as robust as possible, I would have the assert, and would also return NO in the case that it is not a subclass.

In your unittests I would conditionalize the unittest with 

#if !defined(NS_BLOCK_ASSERTIONS)